packages/api-explorer/src/components/ExploreType/ExploreProperty.tsx

```diff
@@ -0,0 +1,262 @@
+  return (
+    <TipIcon
+      show={property.required}
+      icon="Check"
+      tip={tip}
+      title="required property"
+    />
+  )
+}
+
+/**
+ * Show deprecated status if a property is deprecated
+ * @param property to describe
+ * @constructor
+ */
+export const ExplorePropertyDeprecated: FC<ExplorePropertyProps> = ({
+  property,
+}) => {
+  const tip = `${property.fullName} is deprecated`
+  return (
+    <TipIcon
+      show={property.deprecated}
+      icon="Minus"
+      tip={tip}
+      title="deprecated property"
+    />
+  )
+}
+
+/**
+ * Show read-only status if a property is read-only
+ * @param property to describe
+ * @constructor
+ */
+export const ExplorePropertyReadOnly: FC<ExplorePropertyProps> = ({
+  property,
+}) => {
+  const tip = `${property.fullName} is read-only`
+  return (
+    <TipIcon
+      show={property.readOnly}
+      icon="LockClosed"
+      tip={tip}
+      title="read-only property"
+    />
+  )
+}
+
+/**
+ * Display the property description if it's assigned
+ * @param property to describe
+ * @constructor
+ */
+export const ExplorePropertyDescription: FC<ExplorePropertyProps> = ({
+  property,
+}) => {
+  if (!property.description) return <></>
+  return <ReactMarkdown source={property.description} />
+}
+
+/**
+ * Show the details of the property
+ * @param property
+ * @constructor
+ */
+export const ExplorePropertyDetail: FC<ExplorePropertyProps> = ({
+  property,
+}) => (
+  <Space>
+    <FlexItem width="10rem">
+      <ExploreTypeLink type={property.type} />
+    </FlexItem>
+    <FlexItem width="5rem">
+      <ExplorePropertyRequired property={property} />
+      <ExplorePropertyReadOnly property={property} />
+      <ExplorePropertyDeprecated property={property} />
+    </FlexItem>
+    <FlexItem width="30rem">
+      <ExplorePropertyDescription property={property} />
+    </FlexItem>
+  </Space>
+)
+
+/**
+ * Render a potentially complex property as a non-expanding node
+ * @param property to display
+ * @constructor
+ */
+export const ExplorePropertyNode: FC<ExplorePropertyProps> = ({ property }) => {
+  const legend = typeIcon(property.type)
+  return (
+    <TreeItem
+      {...legend}
+      detail={<ExplorePropertyDetail property={property} />}
+    >
+      {property.jsonName}
+    </TreeItem>
+  )
+}
+
+/**
+ * Render the Tree or TreeItem for this property
+ * @param property to display
+ * @param level current nesting level
+ * @param maxDepth maximum depth to expand
+ * @param openAll expands entire tree if true
+ * @constructor
+ */
+export const ExploreProperty: FC<ExplorePropertyProps> = ({
+  property,
+  level = 0,
+  maxDepth = -1,
+  openAll = false,
+}) => {
+  const picked = pickType(property.type)
+  if (!picked.intrinsic) {
+    return (
+      <ExplorePropertyType
+        property={property}
+        open={false}
+        level={level + 1}
+        maxDepth={maxDepth}
+        openAll={openAll}
+      />
+    )
+  }
+  return <ExplorePropertyNode property={property} />
```

Comment:
Thanks to discussions with @joeldodge79 I tend to avoid multiple return statements wherever possible. I know this is a topic of debate but I personally think it improves readability and makes debugging easier when there's a single exit point. Personally, I find that I can see what is going on with a single scan through when this is re-written as below but please feel free to go with whatever you prefer. 
```suggestion
  return (
    <>
      {picked.intrinsic ? (
        <ExplorePropertyNode property={property} />
      ) : (
        <ExplorePropertyType
          property={property}
          open={false}
          level={level + 1}
          maxDepth={maxDepth}
          openAll={openAll}
        />
      )}
    </>
  )
```

Reply:
When I started coding, many languages enforced single return statements and I continued to comply with that constraint for decades. However, I now find the "immediate return" pattern to be clearer and easier to follow than the nested condition suggested.

Reply:
Fair enough :)